### PR TITLE
Issue 332 restructure src

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

Basically I have moved detector modules into a modules folder and made the corresponding cmake changes. No new functionality has been added. I have removed a udpgen_jalousie2 which was nearly identical with udpgen_jalousie. I have aligned cmake files for the modules to look more similar and I have moved TestUDPServer to test/ because it was located in adc_readout but used by other modules too.

The project builds, the tests builds both on my laptop and on Jenkins. There might be, but I can't know in advance, problems with the integration test. At least I am not aware of any.

I would probably not look theough this in github but rather check out the branch and see if the new structure seems resonable.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: cmake/modules/FindIspc.cmake [hunk 1]
```diff
@@ -6,17 +6,17 @@
 # set(myTarget_ISPC_OBJ "")
 # find_package(Ispc)
 # ispc_compile_single_file(myTarget_ISPC_OBJ SampleGen.ispc avx2) # avx2 instruction set support
-# add_executable(myTarget EXCLUDE_FROM_ALL ${myTarget_SRC} ${myTarget_INC} ${myTarget_ISPC_OBJ})
+# add_executable(myTarget ${myTarget_SRC} ${myTarget_INC} ${myTarget_ISPC_OBJ})
```
Comment (MortenHilkerSkaaning) [inline cmake/modules/FindIspc.cmake:9]: note to self: was this reflected in jenkins file

File: documentation/EFU.md [hunk 2]
```diff
@@ -22,10 +22,10 @@ The behaviour of the processing threads is completely determined by the author o
 
 Four examples of already implemented detector interfaces, somewhat alike but also differing are found here
 
-[Gd-GEM](https://github.com/ess-dmsc/event-formation-unit/blob/master/src/gdgem/gdgem.cpp) <br>
-[Multi-Grid](https://github.com/ess-dmsc/event-formation-unit/blob/master/src/multigrid/mgmesytec.cpp) <br>
-[Multi-Blade](https://github.com/ess-dmsc/event-formation-unit/blob/master/src/multiblade/mbcaen.cpp) <br>
-[SoNDe](https://github.com/ess-dmsc/event-formation-unit/blob/master/src/sonde/sonde.cpp)
+[Gd-GEM](https://github.com/ess-dmsc/event-formation-unit/blob/master/src/modules//gdgem/gdgem.cpp) <br>
```
Comment (MortenHilkerSkaaning) [inline documentation/EFU.md:25]: the double slash here looks weird, is this intentional?

File: src/modules/gdgem/srs/CMakeLists.txt [hunk 3]
```diff
@@ -31,11 +31,12 @@ create_test_executable(ParserVMM3Test)
 set(VMM3CalibrationFileTest_SRC
   CalibrationFile.cpp
   CalibrationFileTest.cpp
-  ../../test/SaveBuffer.cpp
+  ${SD}/
```
Comment (MortenHilkerSkaaning) [inline src/modules/gdgem/srs/CMakeLists.txt:34]: maybe SD is a little short here, what does it stand for?

File: documentation/EFU.md (hunk 2 above)
Comment (mortenjc) [reply documentation/EFU.md:25]: nope, a typo

File: src/modules/gdgem/srs/CMakeLists.txt (hunk 3 above)
Comment (mortenjc) [reply src/modules/gdgem/srs/CMakeLists.txt:34]: that is an oversight, I have changed the name to ESS_SOURCE_DIR elswere

File: Jenkinsfile [hunk 4]
```diff
@@ -94,7 +95,7 @@ builders = pipeline_builder.createBuilders { container ->
                 // Ignore file that crashes cppcheck
                 container.sh """
                                 cd ${project}
-                                cppcheck --enable=all --inconclusive --template="{file},{line},{severity},{id},{message}" ./ -isrc/adc_readout/test/SampleProcessingTest.cpp 2> ${test_output}
+                                cppcheck --enable=all --inconclusive --template="{file},{line},{severity},{id},{message}" ./ -isrc/modules//adc_readout/test/SampleProcessingTest.cpp 2> ${test_output}
```
Comment (MortenHilkerSkaaning) [inline Jenkinsfile:98]: typo.

ALSO, we should delete the lines with AdcSimulator and AdcFileStreamer in this file. Github won't allow me to highlight them though. Line 84+226.

**Review by MortenHilkerSkaaning — APPROVED** [review]

**Comment by MortenHilkerSkaaning** [comment]
The new structure looks good. Please merge after you implement the fixes.

File: cmake/modules/FindIspc.cmake (hunk 1 above)
Comment (mortenjc) [reply cmake/modules/FindIspc.cmake:9]: I will remove it from Jenkinsfile

Comment (MortenHilkerSkaaning) [reply cmake/modules/FindIspc.cmake:9]: I meant, AdcSimulator and AdcFileStreamer should be removed as explicit from the jenkinsfile, but we allready have a comment on that :)

File: Jenkinsfile (hunk 4 above)
Comment (MortenHilkerSkaaning) [reply Jenkinsfile]: well, I got feedback from cppcheck on AdcFileStreamer et al, in my PR, so I very much don't think cppcheck crashes...

**Comment by MortenHilkerSkaaning** [comment]
still good to go from me

**Pull request merged**